### PR TITLE
[fixed] issue where edit signing-key used the wrong key to sign the acc

### PR DESCRIPTION
**Pull request by matthiashanel**
Signed-off-by: Matthias Hanel <mh@synadia.com>

```
> nsc describe account A
+--------------------------------------------------------------------------------------+
|                                   Account Details                                    |
+---------------------------+----------------------------------------------------------+
| Name                      | A                                                        |
| Account ID                | ADUQSDJN7Z5DAWUCETP3JUUFSWC5HC5F5GWMZ3FZBXRQ2YW6IA2IUX6U |
| Issuer ID                 | ODX6BDXX554GGX6RKITUEQCP4JEB2X6SKZ6FIP2ZRPUBE7L3AD67ERHH |
| Issued                    | 2021-10-19 22:51:09 UTC                                  |
| Expires                   |                                                          |
+---------------------------+----------------------------------------------------------+
| Signing Keys              | ADOYMJWJSBZ2WP2VMPTDFEPCOHUTSQQKWWNKDI4VPCEEM3THYI33S5OQ |
+---------------------------+----------------------------------------------------------+
| Max Connections           | Unlimited                                                |
| Max Leaf Node Connections | Unlimited                                                |
| Max Data                  | Unlimited                                                |
| Max Exports               | Unlimited                                                |
| Max Imports               | Unlimited                                                |
| Max Msg Payload           | Unlimited                                                |
| Max Subscriptions         | Unlimited                                                |
| Exports Allows Wildcards  | True                                                     |
| Response Permissions      | Not Set                                                  |
+---------------------------+----------------------------------------------------------+
| Jetstream                 | Disabled                                                 |
+---------------------------+----------------------------------------------------------+
| Imports                   | None                                                     |
| Exports                   | None                                                     |
+---------------------------+----------------------------------------------------------+
> nsc edit signing-key --account A --sk ADOYMJWJSBZ2WP2VMPTDFEPCOHUTSQQKWWNKDI4VPCEEM3THYI33S5OQ --role service
[ OK ] edited signing key "ADOYMJWJSBZ2WP2VMPTDFEPCOHUTSQQKWWNKDI4VPCEEM3THYI33S5OQ"
> nsc describe account A
+--------------------------------------------------------------------------------------+
|                                   Account Details                                    |
+---------------------------+----------------------------------------------------------+
| Name                      | A                                                        |
| Account ID                | ADUQSDJN7Z5DAWUCETP3JUUFSWC5HC5F5GWMZ3FZBXRQ2YW6IA2IUX6U |
| Issuer ID                 | ADUQSDJN7Z5DAWUCETP3JUUFSWC5HC5F5GWMZ3FZBXRQ2YW6IA2IUX6U |
| Issued                    | 2021-10-19 22:51:56 UTC                                  |
| Expires                   |                                                          |
+---------------------------+----------------------------------------------------------+
| Signing Keys              | ADOYMJWJSBZ2WP2VMPTDFEPCOHUTSQQKWWNKDI4VPCEEM3THYI33S5OQ |
+---------------------------+----------------------------------------------------------+
| Max Connections           | Unlimited                                                |
| Max Leaf Node Connections | Unlimited                                                |
| Max Data                  | Unlimited                                                |
| Max Exports               | Unlimited                                                |
| Max Imports               | Unlimited                                                |
| Max Msg Payload           | Unlimited                                                |
| Max Subscriptions         | Unlimited                                                |
| Exports Allows Wildcards  | True                                                     |
| Response Permissions      | Not Set                                                  |
+---------------------------+----------------------------------------------------------+
| Jetstream                 | Disabled                                                 |
+---------------------------+----------------------------------------------------------+
| Imports                   | None                                                     |
| Exports                   | None                                                     |
+---------------------------+----------------------------------------------------------+

+---------------------------------------------------------------------------------+
|                          Scoped Signing Key - Details                           |
+----------------------+----------------------------------------------------------+
| Key                  | ADOYMJWJSBZ2WP2VMPTDFEPCOHUTSQQKWWNKDI4VPCEEM3THYI33S5OQ |
| role                 | service                                                  |
| Response Permissions | Not Set                                                  |
| Max Msg Payload      | Unlimited                                                |
| Max Data             | Unlimited                                                |
| Max Subs             | Unlimited                                                |
| Network Src          | Any                                                      |
| Time                 | Any                                                      |
| Bearer Token         | No                                                       |
+----------------------+----------------------------------------------------------+
```

### CODE REVIEW

File: cmd/editscopedsk.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (p *EditScopedSkParams) SetDefaults(ctx ActionCtx) error {
 	if err := p.AccountContextParams.SetDefaults(ctx); err != nil {
 		return err
 	}
-	p.SignerParams.SetDefaults(nkeys.PrefixByteAccount, true, ctx)
+	p.SignerParams.SetDefaults(nkeys.PrefixByteOperator, true, ctx)
```
Comment (aricart) [inline cmd/editscopedsk.go:68]: this works for a local environment. For ngs/etc where the account is signed by the account, this needs to be relaxed to be PrefixByteAccount I think. You can try by updating an NGS account and see that it adds correctly

File: cmd/editscopedsk_test.go [hunk 2]
```diff
@@ -52,11 +53,13 @@ func Test_EditScopedSk_Subs(t *testing.T) {
 	require.NoError(t, err)
 	require.Contains(t, ac.SigningKeys, pk)
 	require.Contains(t, ac.SigningKeys, pk2)
+	require.True(t, nkeys.IsValidPublicOperatorKey(ac.Issuer))
```
Comment (aricart) [inline cmd/editscopedsk_test.go:56]: same issue

File: cmd/editscopedsk_test.go [hunk 3]
```diff
@@ -52,11 +53,13 @@ func Test_EditScopedSk_Subs(t *testing.T) {
...
 
 	checkAcc := func(subs int64) {
 		ac, err = ts.Store.ReadAccountClaim("A")
 		require.NoError(t, err)
 		require.Contains(t, ac.SigningKeys, pk)
+		require.True(t, nkeys.IsValidPublicOperatorKey(ac.Issuer))
```
Comment (aricart) [inline cmd/editscopedsk_test.go:62]: same issue

Comment (matthiashanel) [reply cmd/editscopedsk_test.go:62]: fixed

**Review by aricart — APPROVED** [review]
LGTM

**Pull request merged**